src/zcash/Address.cpp

```diff
@@ -174,6 +192,16 @@ std::string libzcash::UnifiedFullViewingKey::Encode(const KeyConstants& keyConst
     return res;
 }
 
+std::optional<libzcash::OrchardFullViewingKey> libzcash::UnifiedFullViewingKey::GetOrchardKey() const {
+    std::vector<uint8_t> buffer(96);
+    if (unified_full_viewing_key_read_orchard(inner.get(), buffer.data())) {
+        CDataStream ss(buffer, SER_NETWORK, PROTOCOL_VERSION);
+        return OrchardFullViewingKey::Read(ss);
+    } else {
+        return std::nullopt;
+    }
```

Comment:
In this case, is it actually necessary to route through the encoded form? If the UFVK already has a pointer to an Orchard FVK allocated as Rust memory, why not just clone it on the rust side and return that, rather than serialize and then parse? For Sapling and transparent keys the types are all C++ types and the memory is managed by C++, but in the Orchard case `OrchardFullViewingKey` is already just a wrapper around a pointer into Rust-allocated memory.

Reply:
(non-blocking, but maybe worth a change if we're doing another round of CI)

Reply:
Opened #5584

Reply:
This is now #5584.